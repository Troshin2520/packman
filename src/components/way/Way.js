import React, {Component} from 'react';
import PropTypes from 'prop-types';
import './Way.less';

class Way extends Component {

  constructor(props) {
    super(props);
  }

  render() {
    return (<div className={`way ${this.props.has || 'empty'}`}>
      <div></div>
    </div>);
  }
}

Way.propTypes = {
  has: PropTypes.oneOf(['empty', 'feed', 'pill'])
};

export default Way;