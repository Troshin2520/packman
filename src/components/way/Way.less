.way {
  width: 4rem;
  height: 4rem;
  background-color: lightgray;
  display: flex;
  justify-content: center;
  align-items: center;

  &.feed > div {
    background-color: darkmagenta;
    width: 0.5rem;
    height: 0.5rem;
    border-radius: 0.2rem;
  }

  &.pill > div {
    background-color: cornflowerblue;
    width: 1.5rem;
    height: 1.5rem;
    border-radius: 100%;
  }

  &.empty > div {
    background-color: transparent;
  }
}

