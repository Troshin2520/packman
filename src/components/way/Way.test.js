import React from 'react';
import Renderer from 'react-test-renderer';
import Enzyme, {mount} from 'enzyme'
import Adapter from "enzyme-adapter-react-16";
import configureStore from 'redux-mock-store';
import Way from './Way';

Enzyme.configure({ adapter: new Adapter() });

describe('Way testing', () => {

  const initialState = {field: [[4, 0, 4],
                                [0, 1, 2],
                                [4, 4, 4]
                              ]};
  const mockStore = configureStore();
  const store = mockStore(initialState);

  it('Way rendered without crushing', () => {
    const component = Renderer.create(
      <Way store={store} x={0} y={1}  />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Checked default type', () => {
    const wrapper = mount(<Way store={store} x={1} y={0}/>);
    expect(wrapper.find('.way').hasClass('empty')).toBeTruthy();
  });

  it('`pill` attribute added successfully', () => {
    const wrapper = mount(<Way store={store} x={1} y={1}/>);
    expect(wrapper.find('.way').hasClass('feed')).toBeTruthy();
  });

  it('`pill` attribute added successfully', () => {
    const wrapper = mount(<Way store={store} x={2} y={1}/>);
    expect(wrapper.find('.way').hasClass('pill')).toBeTruthy();
  });

});