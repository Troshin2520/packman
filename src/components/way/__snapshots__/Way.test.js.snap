// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Way testing Way rendered without crushing 1`] = `
<div
  className="way empty"
>
  <div />
</div>
`;
